lib/order.rb

```diff
@@ -0,0 +1,76 @@
+class Order
+
+  attr_accessor :complete
+
+  def initialize (menu)
+    @menu = menu.view
+    @order = {}
+    @total = 0.0
+    @complete = false
+
+  end
+
+  def view_order
+    @order.clone
+
+  end
+
+  def view_total
+    copy = @total.to_s
+    copy.to_f
+
+  end
+
+  def add pizza, number
+    raise 'order has already been completed' if complete
+    raise 'item not on menu' if !@menu.include?(pizza)
+    raise 'invalid number' if number % 1 != 0
```

Comment:
These are really nice 'guard conditions' -> they stop any problematic input from progressing further in the program.